### PR TITLE
Added support of using trackingSubmodules in the DSL for git submodules.

### PR DESCRIPTION
**Pull request by zjfroot**
I followed the same implementation of recursiveSubmodules.
trackingSubmodules is the option to keep tracked git submodules updated to tip of branch

### CODE REVIEW

File: job-dsl-core/src/main/groovy/javaposse/jobdsl/dsl/helpers/scm/GitContext.groovy [hunk 1]
```diff
@@ -170,6 +171,13 @@ class GitContext extends AbstractContext {
     }
 
     /**
+     *Keep
```
Comment (daspilker) [inline job-dsl-core/src/main/groovy/javaposse/jobdsl/dsl/helpers/scm/GitContext.groovy]: Add a full sentence.

Comment (daspilker) [inline job-dsl-core/src/main/groovy/javaposse/jobdsl/dsl/helpers/scm/GitContext.groovy]: And add a `@since 1.40` tag.

Comment (zjfroot) [inline job-dsl-core/src/main/groovy/javaposse/jobdsl/dsl/helpers/scm/GitContext.groovy]: Added. :-)

**Comment by daspilker** [comment]
Thanks! I squashed your commits and merged them manually.

**Pull request closed without merging**